### PR TITLE
Partially implemented pyproject.toml file

### LINKED ISSUE #14: Start using pyproject.toml
https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html
https://browniebroke.com/blog/convert-requirements-to-pyproject/
https://setuptools.pypa.io/en/latest/build_meta.html

### PR DESCRIPTION
**Pull request by 9gl**
Implemented  a pyproject.toml for 
[pyproject.toml](https://github.com/9gl/pynalyser/blob/main/pyproject.toml) 

- first attempt at #14 
partially
Checklist:

- [x] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
- [ ] Add or update relevant docs, in the `docs` folder and in code docstring.
- [ ] Add an entry in `CHANGES.md` summarizing the change and linking to the issue and your username.
- [ ] Add `*Version changed*` or `*Version added*` note in any relevant docs and docstring.
- [ ] Run `pytest` and `tox`, no tests failed.

### CODE REVIEW

**Comment by 0dminnimda** [comment]
It's nice to see the progress with the pr!

**Comment by 0dminnimda** [comment]
I also will try to land starter docs asap

**Comment by 9gl** [comment]
Nice

**Comment by 0dminnimda** [comment]
@9gl do you plan on adding more changes or do you want me to review and merge?

**Comment by 9gl** [comment]
You can review and merge.I will create new follow up prs

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+[project]
+name = "pynalyser"
+version = "0.1.0"
+description="Static Python Code Analyzer"
+authors=[
+    {name = "0dminnimda",
+    email="0dminnimda.contact@gmail.com",
+    url="https://github.com/0dminnimda/pynalyser"},
```
Comment (0dminnimda) [inline pyproject.toml:8]: Why this url is in the authors?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+]
+license = {file = "LICENSE "}
```
Comment (0dminnimda) [inline pyproject.toml:10]:
```suggestion
license = {file = "LICENSE"}
```

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+readme = "README.md"
+requires-python = ">=3.7"
+classifiers=[
+        "Development Status :: 2 - Pre-Alpha",
```
Comment (0dminnimda) [inline pyproject.toml:14]: Let's format this to have only one indentation

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        "Programming Language :: Python :: 3.10",
+        "Topic :: Software Development :: Libraries :: Python Modules",
+        "Topic :: Software Development :: Pre-processors",
+        "Topic :: Utilities",
+        "Typing :: Typed",
+]
+
+[project.urls]
+homepage = "https://github.com/0dminnimda/pynalyser"
+documentation = "https://github.com/0dminnimda/pynalyser"
```
Comment (0dminnimda) [inline pyproject.toml:32]: I created the docs, so you can link it

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
+        "Topic :: Software Development :: Libraries :: Python Modules",
+        "Topic :: Software Development :: Pre-processors",
+        "Topic :: Utilities",
+        "Typing :: Typed",
+]
+
+[project.urls]
```
Comment (0dminnimda) [inline pyproject.toml:30]: What about the Bug tracker?

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,33 @@
+[project]
+name = "pynalyser"
+version = "0.1.0"
```
Comment (0dminnimda) [inline pyproject.toml:3]: Let's make version dynamic here and then you can implement this for setuptools in the other pr

**Review by 0dminnimda — CHANGES_REQUESTED** [review]

**Comment by 0dminnimda** [comment]
Oh, yeah, forgot to add
I left minor suggestions for improvement and a question otherwise LGTM.

File: pyproject.toml (hunk 6 above)
Comment (9gl) [reply pyproject.toml:3]: ok

Comment (0dminnimda) [reply pyproject.toml:3]: How is it going? ^-^

Comment (9gl) [reply pyproject.toml:3]: Good 
Will try to complete by next week as I am bit busy currently.

Comment (0dminnimda) [reply pyproject.toml:3]: yep, it's fine, no pressure!

**Comment by 0dminnimda** [comment]
Hey! How is it? 😉

**Comment by 9gl** [comment]
Good . I am on it.

**Comment by 0dminnimda** [comment]
Alrigth, let's finally finish with this, it's better to have at least something merged

If you can't work on this right now, then let's do minimal requreded work and I can finish the rest myself
Let's rename the file, fix the indentation and remove the url (https://github.com/0dminnimda/pynalyser/pull/15#discussion_r935813514)
And then it'll be good to go ^_^

What do you think, @9gl?

**Comment by 0dminnimda** [comment]
Also, I'm planning to move this repo into organisation @enhance-python
Can you give me some feedback on the name? Would be nice!

**Comment by 9gl** [comment]
> Alright, let's finally finish with this, it's better to have at least something merged
> 
> If you can't work on this right now, then let's do minimal requreded work and I can finish the rest myself
> Let's rename the file, fix the indentation and remove the url (https://github.com/0dminnimda/pynalyser/pull/15#discussion_r935813514)
> And then it'll be good to go ^_^
> 
> What do you think, @9gl?

Ok

File: pyproject.toml (hunk 1 above)
Comment (0dminnimda) [inline pyproject.toml:8]:
```suggestion
    email="0dminnimda.contact@gmail.com"},
```

**Comment by 0dminnimda** [comment]
Thanks @9gl!

**Comment by 0dminnimda** [comment]
Also @all-contributors let's add @9gl for infra

**Pull request merged**